scout/server/blueprints/cases/controllers.py

```diff
@@ -66,6 +66,23 @@ def cases(store, case_query, limit=100):
         case_obj['display_track'] = TRACKS[case_obj.get('track', 'rare')]
         case_groups[case_obj['status']].append(case_obj)
 
+    if prioritized_cases_query:
```

Comment:
I might be missing something but why instead of having 2 loops and collecting first all the info for the non-prio cases and then for the prio-case, you don't do 
```list(prioritized_cases_query) + list(case_query)```
and then you loop just once?  This way the prio case will be in the beginning anyway

Reply:
The idea is likely to avoid duplicates. We could just use a set, I suppose, but I'm not super certain how that works on general objects. Lets see!

Reply:
It will again cost a bit more memory, since the objects will first be instantiated in the set, then appended to their case groups. Maybe the case groups could be sets?

Reply:
Ok, thanks, that saved me some writing, and might have been it - or just me chickening out on the whole deal of juggling cursors, lists, dicts, enumerate builtin tuple operation and sets - not to mention extra tuples - in a dynamically typed, but type sensitive language. 😆 

I added a local function instead. This may come across as somewhat procedural code. I for the time being choose to consider it functional, somewhat encapsulated and avoiding the main concern with repeating code. Also I'm a little tired. :wink: